ugvc/mrd/featuremap_consensus_utils.py

```diff
@@ -0,0 +1,323 @@
+        output_vcf (str): The output vcf file
+    """
+    cons_dict = defaultdict(dict)
+    # process genomic interval input
+    if genomic_interval is None:
+        chrom = None
+        start = None
+        end = None
+    else:
+        assert re.match(r"^\w+:\d+-\d+$", genomic_interval), "Input genomic_interval format should be 'chrom:start-end'"
+        genomic_interval_list = genomic_interval.split(":")
+        chrom = genomic_interval_list[0]
+        start = int(genomic_interval_list[1].split("-")[0])
+        end = int(genomic_interval_list[1].split("-")[1])
+
+    # generate a new header
+    orig_header = pysam.VariantFile(featuremap).header
+    header = pysam.VariantHeader()
+    header.add_meta("fileformat", "VCFv4.2")
+    header.filters.add(
+        "SingleRead", None, None, "Aggregated quality above threshold, Only a single read agrees with alternative"
+    )
+    header.filters.add("LowQual", None, None, "Aggregated quality below threshold")
+    for field in fields_to_collect["numeric_array_fields"]:
+        assert field in orig_header.info, f"Field {field} not found in the input featuremap vcf"
+        header.info.add(field, ".", orig_header.info[field].type, orig_header.info[field].description)
+    for field in fields_to_collect["string_list_fields"]:
+        assert field in orig_header.info, f"Field {field} not found in the input featuremap vcf"
+        header.info.add(field, "1", orig_header.info[field].type, orig_header.info[field].description)
+    for field in fields_to_collect["boolean_fields"]:
+        assert field in orig_header.info, f"Field {field} not found in the input featuremap vcf"
+        header.info.add(field, "1", "String", orig_header.info[field].description)
+    for field in fields_to_collect["fields_to_write_once"]:
+        assert field in orig_header.info, f"Field {field} not found in the input featuremap vcf"
+        header.info.add(field, "1", orig_header.info[field].type, orig_header.info[field].description)
+    for field in fields_to_collect["boolean_fields_to_write_once"]:
+        assert field in orig_header.info, f"Field {field} not found in the input featuremap vcf"
+        header.info.add(field, "0", "Flag", orig_header.info[field].description)
+    # exceptions: X_QUAL
+    header.info.add(FeatureMapFields.X_QUAL.value, ".", "Float", "Quality of reads containing this location")
+    header.formats.add(GT, 1, "String", "Genotype")
+    header.formats.add(DP, 1, "Integer", "Read depth")
+    header.formats.add(AD, "R", "Integer", "Allelic depths")
+    header.formats.add(VAF, 1, "Float", "Allele frequency")
+    # copy contigs from original header
+    for contig in orig_header.contigs:
+        header.contigs.add(contig)
+    header.add_sample(sample_name)
+
+    # open an output file
+    out_fh = pysam.VariantFile(output_vcf, "w", header=header)
+
+    # sort the vcf file prior to reading
+    sorted_featuremap = featuremap.replace(".vcf.gz", ".sorted.vcf.gz")
```

Comment:
This is not a safe mechanism, you could be overriding something. Specifically if you get a .vcf file it will try to overwrite itself. Using tempfile is generally best IMO, e.g.
```
import tempfile

with tempfile.NamedTemporaryFile(mode='w+t') as sorted_featuremap:
    # do stuff

```
It's also deleted automatically in the end.

Reply:
it doesn't work this way, so I used a temp directory.